### ISSUE TITLE
[BOOM-5311] No update products cart address delivery with the address delivery of the cart

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#9672]
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5311).

- _**Reporter:**_ thibault
- _**Created at:**_ Fri, 13 Apr 2018 11:58:34 +0200

<p>If a cart contain products that have different address delivery and the shop don't have multishipping enable, the addresses of the products in cart are not reinitialized with the address of the cart.</p>

<p>In PrestaShop v1.6 it was made by Cart->setNoMultishipping() which was call in OrderController.php (function initContent() line 149).</p>

<p>I have seen a bug with a cart that contain products with an deleted address so no carriers are shown and the order can't be completed. The addresses of the product's cart are not reset, so the customer can never order until he empty his cart.</p>
        
* How to reproduce the issue ?

I have seen a bug with a cart that contain products with an deleted address so no carriers are shown and the order can&#39;t be completed. The addresses of the product&#39;s cart are not reset, so the customer can never order until he empty his cart.

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5311#comment-166139).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Fri, 13 Apr 2018 12:41:43 +0200

<p>Hi Thibault,</p>

<p>Do you have the same issue with this ticket: <a href="http://forge.prestashop.com/browse/BOOM-3566" title="When a customer edits an used billing address (address used before for order). no shipping methods are available" class="issue-link" data-issue-key="BOOM-3566">BOOM-3566</a>?</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5311#comment-166148).

- _**Author:**_ thibault
- _**Created at:**_ Fri, 13 Apr 2018 14:04:16 +0200

<p>Hi,</p>

<p>no, i reproduced the process described in this ticket (<a href="http://forge.prestashop.com/browse/BOOM-3566" title="When a customer edits an used billing address (address used before for order). no shipping methods are available" class="issue-link" data-issue-key="BOOM-3566">BOOM-3566</a>) and i don't have the problem.</p>

<p>I think the problem i have encountered is different.</p>

<p>Thanks.</p>

<p>Thibault</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5311#comment-166150).

- _**Author:**_ thibault
- _**Created at:**_ Fri, 13 Apr 2018 14:10:45 +0200

<p>Quick fix for this :</p>

<p>in controllers/front/CartController.php, add these lines in the function initContent()</p>

<p> <br/>
if (!Tools::getValue('multi-shipping')) {<br/>
    $this->context->cart->setNoMultishipping();<br/>
}<br/>
 <br/>
(this code is the code found on PS 1.6 on the controller/front/OrderController.php > function initContent())</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5311#comment-166151).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Fri, 13 Apr 2018 14:11:40 +0200

<p>Hi Thibault,</p>

<p>Would you be willing to make a pull request on GitHub with your code suggestion?<br/>
<a href="https://github.com/PrestaShop/PrestaShop/tree/develop" class="external-link" rel="nofollow">https://github.com/PrestaShop/PrestaShop/tree/develop</a><br/>
Thank you!</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5311#comment-173695).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Wed, 8 Aug 2018 12:01:26 +0200

<p>Hi Thibault,</p>

<p>I manage to reproduce the issue with the last version 1.7.4.2.<br/>
I attached a video record.<br/>
 <span class="nobr"><a href="http://forge.prestashop.com/secure/attachment/70240/70240_Aug+8+2018+11_59+AM.webm" title="Aug 8 2018 11_59 AM.webm attached to BOOM-5311">Aug 8 2018 11_59 AM.webm<sup><img class="rendericon" src="http://forge.prestashop.com/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span> <br/>
We will see how to fix it.</p>

<p>Best regards, Khouloud</p>

**Comment by netdrag** [comment]
We are hitting this once or twice a day now on 1.7.5. Don't want to know how many customers we loose because of this. We just know of the ones that contact us. I don't think thats a MINOR Issue.

**Comment by khouloudbelguith** [comment]
Hi @netdrag,

Sorry, it is not fixed yet.
I manage to reproduce the issue with PS1.7.5.0 also.
There are some major issues to solve before this one.
So, it will probably not solved right now. 
But PrestaShop is an open source project, so it can be solved before if someone submits a pull request to solve it.
Thanks for your understanding!

**Comment by marionf** [comment]
Related to #9932

**Comment by Hlavtox** [comment]
No longer relevant, cart_product id_address_delivery is no longer used on develop.
